pkg/cmd/dedicated/deregister/deregister.go

```diff
@@ -117,15 +118,12 @@ func getListOfClusters(opts *options) ([]*clustersmgmtv1.Cluster, error) {
 	if err != nil {
 		if response != nil {
 			if response.StatusCode == 403 {
-				return nil, opts.f.Localizer.MustLocalizeError("")
+				return nil, opts.f.Localizer.MustLocalizeError("dedicated.deregisterCluster.error.403")
 			}
 
-			return nil, fmt.Errorf("%v, %v", response.Status, err)
+			return nil, fmt.Errorf("%v, %w", response.Status, err)
 		}
-
-		return nil, err
```

Comment:
You need `return err` inside the `err != nil` check and outside of the `response != nil` check